<a href="https://colab.research.google.com/github/repatoon/sergin/blob/4semestr/_4_ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Easy tasks

Напишите цикл `while` , который позволяет пользователю ввести число. К числу необходимо прибавлять 10 и результат – присваивать переменной с именем product. Цикл должен повторяться до тех пор, пока product меньше 100.

In [ ]:
product = 0

while product < 100:
    try:
        number = int(input("Введите число: "))
        product = number + 10
        print(f"Текущее значение product: {product}")
    except ValueError:
        print("Ошибка: введите целое число")

print("Цикл завершен")

Напишите цикл `while`, который просит пользователя ввести два числа. Числа должны быть сложены, и показана сумма. Цикл должен запрашивать у пользователя, желает ли он выполнить операцию еще раз. Если да, то цикл должен повториться, в противном случае он должен завершиться.

In [ ]:
repeat = 'д'
while repeat.lower() == 'д':
    num1 = float(input("Введите первое число: "))
    num2 = float(input("Введите второе число: "))
    total = num1 + num2
    print(f"Сумма чисел: {total}")
    repeat = input("Хотите продолжить? (д/н): ")
print("Программа завершена.")

Введите первое число: 4
Введите второе число: 5
Сумма чисел: 9.0
Хотите продолжить? (д/н): y
Программа завершена.


Напишите цикл for, который выводит приведенный ниже ряд чисел:

0, 10, 20, 30, 40, 50, ..., 1000

In [ ]:
for number in range(0, 1001, 10):
    print(number, end=', ' if number < 1000 else '')

0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260, 270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370, 380, 390, 400, 410, 420, 430, 440, 450, 460, 470, 480, 490, 500, 510, 520, 530, 540, 550, 560, 570, 580, 590, 600, 610, 620, 630, 640, 650, 660, 670, 680, 690, 700, 710, 720, 730, 740, 750, 760, 770, 780, 790, 800, 810, 820, 830, 840, 850, 860, 870, 880, 890, 900, 910, 920, 930, 940, 950, 960, 970, 980, 990, 1000

Напишите цикл, который просит пользователя ввести число. Цикл должен выполнить 10 итераций и вести учет нарастающего итога введенных чисел.

In [ ]:
total = 0
for i in range(1, 11):
    while True:
        try:
            number = float(input(f"Введите число {i} из 10: "))
            break
        except ValueError:
            print("Ошибка! Введите число.")
    total += number
    print(f"Текущий итог: {total}")

print(f"\nОбщая сумма: {total}")

Введите число 1 из 10: 5
Текущий итог: 5.0


# More tasks

*1 балл*

**Сожженные калории.** Бег на беговой дорожке позволяет сжигать 4,2 калорий в минуту. Напишите программу, которая применяет цикл для вывода количества калорий, сожжен­ ных после 10, 15, 20, 25 и 30 минут бега.

In [ ]:

calories_per_minute = 4.2
minutes = [10, 15, 20, 25, 30]
print("Минуты\tКалории")
print("----------------")
for time in minutes:
    calories_burned = calories_per_minute * time
    print(f"{time}\t{calories_burned:.1f}")

*1 балл*


**Уровень океана.** Допустим, что уровень океана в настоящее время повышается пример­но на 1,6 мм в год. С учетом этого создайте приложение, которое выводит количество миллиметров, на которые океан будет подниматься каждый год в течение следующих 25 лет.


In [ ]:

rise_per_year = 1.6
total_years = 25
print("Год\tПовышение уровня (мм)")
print("--------------------------")
for year in range(1, total_years + 1):
    total_rise = rise_per_year * year
    print(f"{year}\t{total_rise:.1f}")

*1 балл*

**Пройденное расстояние.** Пройденное транспортным средством расстояние можно вы­ числить следующим образом:

расстояние = скорость х время.

Например, если поезд движется со скоростью 90 км/ч в течение трех часов, то пройден­ное расстояние составит 270 км. Напишите программу, которая запрашивает у пользова­теля скорость транспортного средства (в км/ч) и количество часов, которое оно двига­лось. Затем она должна применить цикл для вывода расстояния, пройденного транспорт­ным средством для каждого часа этого периода времени.

Вот пример требуемого результата:
```
Какая скорость транспортного средства в км/ч? 40
Сколько часов оно двигалось? 3
Hour | Distance
---------------------
1    | 40
2    | 80
3    | 120
```

In [ ]:

speed = int(input("Какая скорость транспортного средства в км/ч? "))
hours = int(input("Сколько часов оно двигалось? "))
print("\nHour | Distance")
print("---------------------")
for hour in range(1, hours + 1):
    distance = speed * hour
    print(f"{hour:4} | {distance:5}")

*1 балл*

**Средняя толщина дождевых осадков.** Напишите программу, которая применяет *вло­женные циклы* для сбора данных и вычисления средней толщины дождевых осадков за ряд лет. Программа должна сначала запросить количество лет. Внешний цикл будет вы­полнять одну итерацию для каждого года. Внутренний цикл будет делать двенадцать итераций, одну для каждого месяца. Каждая итерация внутреннего цикла запрашивает у пользователя миллиметры дождевых осадков в течение этого месяца. После всех итера­ций программа должна вывести количество месяцев, общее количество миллиметров до­ждевых осадков и среднюю толщину дождевых осадков в месяц в течение всего периода.




*1 балл*

**Рост платы за обучение.** В заданном университете обучение студента-очника составля­ ет 45 000 рублей в семестр. Было объявлено, что плата за обучение будет повышаться на 3% каждый год в течение следующих 5 лет. Напишите программу с циклом, который выводит плановую сумму за обучение в семестр в течение следующих 5 лет.

*1 балл*

**Потеря массы.** Если умеренно активный человек будет сокращать свое потребление в калориях на 500 калорий в день, то, как правило, он может похудеть примерно на 1,5 кг в месяц. Напишите программу, которая позволяет пользователю ввести его исход­ную массу и затем создает и выводит таблицу, показывающую, каким будет его ожидае­мая масса в конце каждого месяца в течение следующих 6 месяцев, если он останется на этой диете.

*1 балл*

**Популяция.** Напишите программу, которая предсказывает приблизительный размер по­пуляции организмов. Приложение должно дать пользователю ввести стартовое количество организмов, среднесуточное увеличение популяции (как процент) и количество дней, которые организмам будет дано на размно­жение. Например, допустим, что пользователь вводит приведенные ниже значения:
- стартовое количество: 2;
- среднесуточное увеличение: 30%;
- количество дней для размножения: 10.

Программа должна вывести показанную ниже таблицу данных:
```
Day  Population
1    2
2    2,6
3    3,38
4    4,394
5    5,7122
6    7,42586
7    9,653619
8    12,5497
9    16,31462
10   21,209
```

*2 балла*

**Палиндром.** Строка называется палиндромом, если она пишется одинаково в обоих направлениях. Например, палиндромами в английском языке являются слова «anna», «civic», «level», «hannah». Напишите программу, запрашивающую у пользователя строку и при помощи цикла определяющую, является ли она палиндромом. Задачу необходимо решить с использованием цикла.

*2 балла*

**Код Цезаря**

Одним из первых в истории примеров шифрования считаются закодиро
ванные послания Юлия Цезаря. Римскому полководцу необходимо было
посылать письменные приказы своим генералам, но он не желал, чтобы в случае чего их прочитали недруги. В результате он стал шифровать свои послания довольно простым методом, который впоследствии стали называть кодом Цезаря.

Идея шифрования была совершенно тривиальной и заключалась в циклическом сдвиге букв на три позиции. В итоге буква A превращалась в D, B – в E, C – в F и т. д. Последние три буквы алфавита переносились на начало. Таким образом, буква X становилась A, Y – B, а Z – C. Цифры и другие
символы не подвергались шифрованию.

Напишите программу, реализующую код Цезаря. Позвольте пользователю ввести фразу и количество символов для сдвига, после чего выведите результирующее сообщение. Убедитесь в том, что ваша программа шифрует как строчные, так и прописные буквы. Также должна быть возможность указывать отрицательный сдвиг, чтобы можно было использовать вашу программу для расшифровки фраз.

*3 балла*

**Подбрасываем монетку.** Какое минимальное количество раз вы должны подбросить монетку, чтобы три раза подряд выпал либо орел, либо решка? А какое максимальное количество попыток может для этого понадобиться? А в среднем? В данном упражнении мы выясним это, создав симулятор подбрасывания виртуальной монетки.

Напишите программу, использующую для подброса монетки генера-
тор случайных чисел Python. Монетка при этом должна быть правильной формы, что означает равную вероятность выпадения орла и решки. Подбрасывать монетку необходимо до тех пор, пока три раза подряд не выпадет одно значение, вне зависимости от того, орел это будет или решка. Выводите на экран букву О всякий раз, когда выпадает орел, и Р – когда выпадает решка. При этом для одной симуляции бросков все выпавшие значения необходимо размещать на одной строке. Также необходимо известить пользователя о том, сколько попыток потребовалось, чтобы получить нужный результат.

Программа должна выполнить десять симуляций и в конце представить среднее количество подбрасываний монетки, требуемое для достижения нужного нам результата. Пример вывода программы показан ниже:

```
О Р Р Р (попыток: 4)
О О Р Р О Р О Р Р О О Р О Р Р О Р Р Р (попыток: 19)
Р Р Р (попыток: 3)
Р О О О (попыток: 4)
О О О (попыток: 3)
Р О Р Р О Р О О Р Р О О Р О Р О О О (попыток: 18)
О Р Р О О О (попыток: 6)
Р О Р Р Р (попыток: 5)
Р Р О Р Р О Р О Р О О О (попыток: 12)
Р О Р Р Р (попыток: 5)
Среднее количество попыток: 7,9.
```